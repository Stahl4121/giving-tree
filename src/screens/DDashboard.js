import React, { useEffect } from 'react';
import SmallGrantCard from '../components/SmallGrantCard.js';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import { useCollection } from 'react-firebase-hooks/firestore';
import firebase from '../firebase.js';
import Search from '../components/Search';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  container: {
    paddingBottom: theme.spacing(2),
  },
}));

export default function DDashboard() {
  // List of small grant cards
  const [grants, setGrants] = React.useState([]);

  // Initialize database and specific grant in database
  const db = firebase.firestore();

  const [snapshot, loading, error] = useCollection(db.collection('grants'));

  useEffect(() => {
    var newGrants = [];
    if (!loading && !error) {
      snapshot.forEach(function (doc) {
        newGrants.push(
          <SmallGrantCard
            id={doc.id}
            title={doc.data().title}
            cfName={doc.data().cf_name}
            nonprofitName={doc.data().nonprofit_name}
            goalAmt={doc.data().goal_amt}
            moneyRaised={doc.data().money_raised}
            img={doc.data().images[0] || 'GivingTree.png'} />
        );
      });
      setGrants(newGrants);
    }
  }, [snapshot, error, loading]);
  /*
  useEffect(() => {
    var useless = 0;
  }, [grants]);
  */
  function searchCallback(childData) {
    var newGrants = [];
    childData.forEach((grant) => {
      newGrants.push(
        <SmallGrantCard
          id={grant.id}
          title={grant.title}
          cfName={grant.cf_name}
          nonprofitName={grant.nonprofit_name}
          goalAmt={grant.goal_amt}
          moneyRaised={grant.money_raised}
          img={grant.images[0] || 'GivingTree.png'} />
      );
    });
    setGrants(newGrants);
  }

  const classes = useStyles();

  return (
    <Container maxWidth='md' className={classes.container}>
      <Search parentCallback={searchCallback} />
      <Grid container spacing={2} >
        {grants}
      </Grid>
    </Container>
  );
}
