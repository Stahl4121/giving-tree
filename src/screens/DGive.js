// https://stripe.com/docs/recipes/elements-react

import React, { useEffect } from 'react';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/styles';
import Text from '../components/Text.js';
import firebase from '../firebase.js';
import Snack from '../components/Snack.js';
import {
  CardElement,
  injectStripe
} from 'react-stripe-elements';
import Button from '@material-ui/core/Button';

//const FieldValue = require('firebase-admin').firestore.FieldValue;

const useStyles = makeStyles(theme => ({
  pageLayout: {
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
      width: 600,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
    marginTop: theme.spacing(8),
    marginBottom: theme.spacing(2),
  },
  paymentPaper: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
    padding: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
      marginTop: theme.spacing(6),
      marginBottom: theme.spacing(6),
      padding: theme.spacing(3),
    },
  },
  stripeElement: {
    display: 'block',
    margin: '10px 0 20px 0',
    width: '100%',
    padding: '10px 14px',
    fontSize: '1em',
    fontFamily: 'Roboto',
    boxShadow: 'rgba(50, 50, 93, 0.14902) 0px 1px 3px, rgba(0, 0, 0, 0.0196078) 0px 1px 0px',
    border: '0',
    outline: '0',
    borderRadius: '4px',
    background: 'white',
  },
}))


function Stripe(props) {
  const classes = useStyles();

  const db = firebase.firestore();

  const [complete, setComplete] = React.useState(false);
  const [amount, setAmount] = React.useState('');
  const [grantName] = React.useState('Test Grant'); //React.useState(props.grantName);
  const [given, setGiven] = React.useState('');
  const [goal, setGoal] = React.useState('');
  const [grantID] = React.useState('wJArP9RCeQY9vDvsfIA2') //React.useState(props.grantID);

  const docRef = db.collection('grants').doc(grantID);


  useEffect(() => {
    docRef.onSnapshot((doc) => {
      if (doc.exists) {
        setGoal(doc.data().goal_amt);
        setGiven(doc.data().money_raised);
      } else {
        console.log('No such grant!');
      }
    })
  });

  const submit = async (ev) => {
    let { token } = await props.stripe.createToken({ name: 'Giving Tree Donor' });
    let response = await fetch('/charge', {
      method: 'POST',
      headers: { 'Content-Type': 'text/plain' },
      body: token.id + ' amount: ' + (amount * 100) + ' description: ' + grantName,
    });

    if (response.ok) {
      // docRef.update({
      //   money_raised: firebase.firestore.FieldValue.increment(amount)
      // });
      db.runTransaction(function (transaction) {
        // This code may get re-run multiple times if there are conflicts.
        return transaction.get(docRef).then(function (doc) {
          if (!doc.exists) {
            throw new Error("Document does not exist!");
          }

          // Note: this could be done without a transaction by updating the population using FieldValue.increment()
          var newMoneyRaised = doc.data().money_raised + Number.parseInt(amount);
          transaction.update(docRef, { money_raised: newMoneyRaised });
        });
      }).then(function () {
        setComplete(true);
      }).catch(function (error) {
        console.log("Transaction failed: ", error);
      });
    }
  }

  return (
    <Container className={classes.pageLayout}>
      <React.Fragment>
        <Paper elevation={3} className={classes.paymentPaper}>
          <Text type='card-heading' text={grantName} />
          {complete ?
            <Text type='card-subheading' text={'Thank you for your donation! Thanks to your gift of $' + amount + ', ' + grantName + ' is now only  $' + (goal - given) + ' from meeting its goal of $' + goal + '!'} />
            :
            <div>
              <Text type='card-subheading' text={'So far, $' + given + ' has been raised out of $' + goal} />
              <CardElement className={classes.stripeElement} />
              <input
                className={classes.stripeElement}
                placeholder="Amount"
                onInput={e => setAmount(e.target.value)} />
              <Button
                color="primary"
                className={classes.button}
                variant="contained"
                onClick={submit}>
                Donate
              </Button>
            </div>
          }
        </Paper>
      </React.Fragment>
    </Container>
  );
}

export default injectStripe(Stripe);