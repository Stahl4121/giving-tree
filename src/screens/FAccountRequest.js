import React from 'react';

import Text from '../components/Text.js';

import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  marginBottom: {
    marginBottom: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  links: {
    textDecoration: 'none'
  },
  errorMsg: {
    color: 'red',
    marginTop: theme.spacing(1),
  },
}));


function FAccountRequest() {
  const classes = useStyles();

  return (
    <Container component="main" maxWidth="md" >
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Text type='card-heading' text={'Foundation Account Request'} />
        <form className={classes.form} noValidate>
          <Grid container spacing={2}>

          <Container maxWidth="xs" className={classes.marginBottom}>
            <Grid container spacing={2}>
              <Grid container item xs={12}>
                <Text type='card-sectionheading' text={'Foundation Info'} />
                <Text type='card-subheading' text={'This information will be displayed publicly on all published giving opportunities.'} />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  name="name"
                  variant="outlined"
                  required
                  fullWidth
                  id="name"
                  label="Foundation Name"
                  autoFocus
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  name="foundation_url"
                  variant="outlined"
                  required
                  fullWidth
                  id="foundation_url"
                  label="Website URL"
                  placeholder="www.communityfoundation.com"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="public_email"
                  variant="outlined"
                  required
                  fullWidth
                  id="public_email"
                  label="Public Email"
                  placeholder="contact@cf.com"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="public_phone"
                  variant="outlined"
                  required
                  fullWidth
                  id="public_phone"
                  label="Public Phone"
                />
              </Grid>
            </Grid>
            </Container>

            <Container maxWidth="xs">
            <Grid container spacing={2}>
              <Grid container item xs={12}>
                <Text type='card-sectionheading' text={'Personal Contact Info'} />
                <Text type='card-subheading' text={'This information will be used to sign into and manage your account.'} />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="fname_contact"
                  variant="outlined"
                  required
                  fullWidth
                  id="fname_contact"
                  label="First Name"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="lname_contact"
                  label="Last Name"
                  name="lname_contact"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="personal_email"
                  label="Personal Email"
                  name="personal_email"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="personal_phone"
                  variant="outlined"
                  required
                  fullWidth
                  id="personal_phone"
                  label="Personal Phone"
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  name="passwordOne"
                  label="Password"
                  type="password"
                  id="passwordOne"
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  name="passwordTwo"
                  label="Confirm Password"
                  type="password"
                  id="passwordTwo"
                />
              </Grid>
            </Grid>
            </Container>

            
            <Container maxWidth="xs">
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Request Account
          </Button>
            <Grid container justify="flex-end">
              <Grid item>
                <Link href="/signin" variant="body2">
                  Already have an account? Sign in
              </Link>
              </Grid>
            </Grid>
            </Container>

          </Grid>
        </form>
      </div>
    </Container>
  );
}

export default FAccountRequest;