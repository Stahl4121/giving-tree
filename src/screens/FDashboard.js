import React, { useEffect, useContext } from 'react';
import SmallGrantCard from '../components/SmallGrantCard.js';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import { useCollection } from 'react-firebase-hooks/firestore';
import firebase from '../firebase.js';
import Search from '../components/Search';
import { makeStyles } from '@material-ui/core/styles';
import UserAuthContext from '../auth/context.js';
import ToggleButton from '@material-ui/lab/ToggleButton';
import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';

const useStyles = makeStyles(theme => ({
  container: {
    paddingBottom: theme.spacing(2),
  },
  toggleBar: {
    marginBottom: 4,
  }
}));

export default function FDashboard(props) {

  const classes = useStyles();

  // List of small grant cards
  const [grants, setGrants] = React.useState([]);
  const [docs, setDocs] = React.useState();
  const [expiredGrants, setExpiredGrants] = React.useState([]);
  const [currentGrants, setCurrentGrants] = React.useState([]);
  const [draftedGrants, setDraftedGrants] = React.useState([]);
  const [toggleBarStatus, setToggleBarStatus] = React.useState('current');

  // Foundation ID
  const user = useContext(UserAuthContext);

  // Initialize database and specific grant in database
  const db = firebase.firestore();
  const [snapshot, loading, error] = useCollection(db.collection('grants'));

  useEffect(() => {
    let newDocs = [];
    let curGrants = [];
    let exGrants = [];
    let drGrants = [];
    if (!loading && !error) {
      snapshot.forEach(function (doc) {
        if (doc.data().cf_id === user?.cfId) {
          newDocs.push({
            dist: -1,
            id: doc.id,
            title: doc.data().title,
            cfName: doc.data().cf_name,
            nonprofitName: doc.data().nonprofit_name,
            goalAmt: doc.data().goal_amt,
            moneyRaised: doc.data().money_raised,
            img: doc.data().images[0] || 'GivingTree.png',
            nonprofitId: doc.data().nonprofit_id,
            address: doc.data().address,
            lat: doc.data().lat,
            long: doc.data().long,
            datePosted: doc.data().date_posted,
            dateDeadline: doc.data().date_deadline,
            desc: doc.data().desc,
            tags: doc.data().tags,
            status: doc.data().status
          });

          let grant = <SmallGrantCard
            id={doc.id}
            title={doc.data().title}
            cfName={doc.data().cf_name}
            nonprofitName={doc.data().nonprofit_name}
            goalAmt={doc.data().goal_amt}
            moneyRaised={doc.data().money_raised}
            img={doc.data().images[0] || 'GivingTree.png'}
            status={doc.data().status} />;
          if (doc.data().status === 'current') {
            curGrants.push(grant);
          } else if (doc.data().status === 'draft') {
            drGrants.push(grant);
          } else if (doc.data().status === 'expired') {
            exGrants.push(grant);
          }
        }
      });
      setDocs(newDocs);
      setCurrentGrants(curGrants);
      setDraftedGrants(drGrants);
      setExpiredGrants(exGrants);
    }
  }, [snapshot, error, loading]);

  function searchCallback(childData) {
    // console.log("childData in dashboard: ", childData);
    var curGrants = [];
    var exGrants = [];
    var drGrants = [];
    childData.forEach((meta) => {
      let grant = <SmallGrantCard
        id={meta.grant.id}
        title={meta.grant.title}
        cfName={meta.grant.cfName}
        nonprofitName={meta.grant.nonprofitName}
        goalAmt={meta.grant.goalAmt}
        moneyRaised={meta.grant.moneyRaised}
        img={meta.grant.img} />
      if (meta.grant.status === 'current') {
        curGrants.push(grant);
      } else if (meta.grant.status === 'draft') {
        drGrants.push(grant);
      } else if (meta.grant.status === 'expired') {
        exGrants.push(grant);
      }
    });
    console.log("CURRENT GRANTS: " + curGrants);

    setCurrentGrants(curGrants);
    setDraftedGrants(drGrants);
    setExpiredGrants(exGrants);
  }

  const handleToggle = (event, status) => {
    setToggleBarStatus(status);
  };

  const buttonOptions = [
    <ToggleButton key={1} value="current">
      Current
    </ToggleButton>,
    <ToggleButton key={2} value="drafted">
      Drafted
    </ToggleButton>,
    <ToggleButton key={3} value="expired">
      Expired
    </ToggleButton>
  ];

  return (
    <Container maxWidth='md' className={classes.container}>
      {docs &&
        <div>
          <Grid container spacing={2} direction="column" alignItems="center">
            <Grid item className={classes.toggleBar}>
              <ToggleButtonGroup size="small" value={toggleBarStatus} exclusive onChange={handleToggle}>
                {buttonOptions}
              </ToggleButtonGroup>
            </Grid>
          </Grid>
          <Search docs={docs} parentCallback={searchCallback} />
          <Grid container spacing={2} >
            {toggleBarStatus === 'current' ? currentGrants : toggleBarStatus === 'drafted' ? draftedGrants : expiredGrants}
          </Grid>
        </div>
      }
    </Container>
  );
}
