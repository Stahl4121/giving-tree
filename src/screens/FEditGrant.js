import React from 'react';
export default function FEditGrant() {
  return(
    <p>Foundation Edit Grant!</p>
  );
}