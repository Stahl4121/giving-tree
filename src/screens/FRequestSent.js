import React, { useEffect, useContext } from 'react';
import { withRouter } from 'react-router-dom';

import { AuthUserContext } from '../auth';
import { withAuthProtection } from '../auth';

import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  message: {
    padding: theme.spacing(8, 0, 6),
  },
}));


function FRequestSent() {
  const classes = useStyles();

  const authUser = useContext(AuthUserContext);

  useEffect(() => {
    if(authUser.accepted){
      this.props.history.push('/foundation');
    }
  }, [authUser]);

  return (
    <Container maxWidth="sm" component="main" className={classes.message}>
        <Typography component="h1" variant="h2" align="center" color="textPrimary" gutterBottom>
          Foundation Account Requested
        </Typography>
        <Typography variant="h5" align="center" color="textSecondary" component="p">
          We have received your request for a foundation account and will be reviewing it shortly. Please check back later.
        </Typography>
      </Container>
  );
}


const condition = (authUser) => authUser?.cf;

export default withAuthProtection(condition)(withRouter(FRequestSent));