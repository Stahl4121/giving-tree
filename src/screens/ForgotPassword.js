import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

import firebase from '../firebase.js';
import * as helper from '../helpers/SignUpHelper.js';
import Snack from '../components/Snack.js';

import { withStyles } from '@material-ui/styles';
import Container from '@material-ui/core/Container';
import Avatar from '@material-ui/core/Avatar';
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Grid from "@material-ui/core/Grid";
import Link from "@material-ui/core/Link";
import Typography from "@material-ui/core/Typography";


const styles = theme => ({
    root: {
        marginTop: theme.spacing(8),
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
    avatar: {
        margin: theme.spacing(1),
        backgroundColor: theme.palette.secondary.main,
    },
    form: {
        marginTop: theme.spacing(1),
    },
    submit: {
        margin: theme.spacing(3, 0, 2),
    },
    links: {
        textDecoration: 'none'
    },
    errorMsg: {
        color: 'red',
        marginTop: theme.spacing(1),
    }
});

function ForgotPassword(props) {
    const { classes } = props;

    return (
        <Container component="main" maxWidth="xs">
            <div className={classes.root}>
                <Avatar className={classes.avatar}>
                    <LockOutlinedIcon />
                </Avatar>
                <Typography component="h1" variant="h5" className={classes.header}>
                    Reset Password
                    </Typography>
                <ForgotPasswordForm classes={classes} />
                <SignInSignUpLink classes={classes} />
            </div>
        </Container>
    );
}

const INITIAL_STATE = {
    email: '',
    emailError: '',
    submitError: '',
    success: false,
};

class ForgotPasswordFormBase extends Component {
    constructor(props) {
        super(props);
        this.state = { ...INITIAL_STATE };
    }

    onSubmit = event => {
        let emailError = helper.validateEmail(this.state.email);

        if (emailError === '') {
            const { email } = this.state;

            firebase.auth().sendPasswordResetEmail(email)
                .then(() => {
                    // Email sent.
                    this.setState({ ...INITIAL_STATE, success: true });
                }).catch((error) => {
                    // An error happened.
                    this.setState({ submitError: error.message, success: false });
                });
        }
        else {
            //Display field validation errors
            this.setState({ emailError: emailError, success: false });
        }

        event.preventDefault();
    }

    onChange = event => {
        this.setState({ [event.target.name]: event.target.value });
    };

    render() {
        const {
            email,
            emailError,
            submitError,
            success,
        } = this.state;

        const isInvalid = (email === '');

        const { classes } = this.props;

        return (
            <form className={classes.form} onSubmit={this.onSubmit} noValidate>
                {success &&
                    <Snack message='Success! A password reset email has been sent.' />
                }
                <Typography component="h6" className={classes.errorMsg} >
                    {submitError}
                </Typography>
                <TextField
                    variant="outlined"
                    margin="normal"
                    required
                    fullWidth
                    name="email"
                    value={email}
                    onChange={this.onChange}
                    type="text"
                    label="Email Address"
                    error={emailError !== ""}
                    helperText={emailError}
                    autoFocus
                />
                <Button className={classes.submit}
                    type="submit"
                    fullWidth
                    variant="contained"
                    color="primary"
                    disabled={isInvalid}
                >
                    Reset Password
                        </Button>
            </form>
        );
    }
}
const SignInSignUpLink = ({ classes }) => (    
    <Grid container>
        <Grid item xs>
            <Link href='/signin' variant="body2">
                    Ready to sign in?
            </Link>
        </Grid>
        <Grid item>
            <Link href='/signup' variant="body2">
                Don't have an account? Sign Up
            </Link>
        </Grid>
    </Grid>
);

const ForgotPasswordForm = withRouter(ForgotPasswordFormBase);

export default withStyles(styles)(ForgotPassword);

export { ForgotPasswordForm, SignInSignUpLink };