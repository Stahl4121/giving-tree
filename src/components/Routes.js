import React from "react"
import { Switch, Route } from "react-router-dom"
import DDashboard from '../screens/DDashboard.js';
import DGive from '../screens/DGive.js';
import DGrant from '../screens/DGrant.js';
import FAccountRequest from '../screens/FAccountRequest.js';
import FDetailedGrant from '../screens/FDetailedGrant.js';
import FEditGrant from '../screens/FEditGrant.js';
import FGrants from '../screens/FGrants.js';
import FRequestSent from '../screens/FRequestSent.js';
import FSettings from '../screens/FSettings.js';
import LInputNewPassword from '../screens/LInputNewPassword.js';
import Login from '../screens/Login.js';
import LRequestNewPassword from '../screens/LRequestNewPassword.js';
import MetaAdmin from '../screens/MetaAdmin.js';

export default function Routes() {
    return (
        <Switch>
            <Route exact path='/' component={DDashboard} />
            <Route exact path='/grants/:grantId' component={DGrant} />
            <Route exact path='/grants/:grantId/give' component={DGive} />
            <Route exact path='/request-account' component={FAccountRequest} />
            <Route exact path='/request-sent' component={FRequestSent} />
            <Route exact path='/foundation' component={FGrants} />
            <Route exact path='/foundation/:grantId' component={FDetailedGrant} />
            <Route exact path='/foundation/:grantId/edit' component={FEditGrant} />
            <Route exact path='/foundation/Settings' component={FSettings} />
            <Route exact path='/login' component={Login} />
            <Route exact path='/request-new-password' component={LRequestNewPassword} />
            <Route exact path='/input-new-password' component={LInputNewPassword} />
            <Route exact path='/meta-admin' component={MetaAdmin} />
            {/*<Route component={NotFound} />*/}
        </Switch>
    )

}