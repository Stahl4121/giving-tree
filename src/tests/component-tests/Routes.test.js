import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router';
import { createMemoryHistory } from "history";

import Routes from '../../components/Routes.js';
import DDashboard from '../../screens/DDashboard.js';
import { DGive } from '../../screens/DGive.js';
import { Grant } from '../../screens/Grant.js';
import { FAccountRequest } from '../../screens/FAccountRequest.js';
import { FEditGrant } from '../../screens/FEditGrant.js';
import { FDashboard } from '../../screens/FDashboard.js';
import { FRequestSent } from '../../screens/FRequestSent.js';
import { FSettings } from '../../screens/FSettings.js';
import { SignIn } from '../../screens/SignIn.js';
import { SignUp } from '../../screens/SignUp.js';
import { ForgotPassword } from '../../screens/ForgotPassword.js';
import { MetaAdmin } from '../../screens/MetaAdmin.js';
import { FStripeSetup } from '../../screens/FStripeSetup.js';

jest.mock('firebase/app');
jest.mock("react-router-dom/BrowserRouter", () => {
    return ({ children }) => <div>{children}</div>;
});

describe('Component: Routes', () => {
    var enzyme = require('enzyme');
    enzyme.configure({ adapter: new Adapter() });

    it('invalid path should not go to any page', () => {
        const wrapper = enzyme.mount(
            <MemoryRouter initialEntries={['/random']}>
                {Routes}
            </MemoryRouter>
        );
        console.log(wrapper.find('Route').at(0).debug());
        expect(wrapper.find(DDashboard)).toHaveLength(0);
        expect(wrapper.find('DGive')).toHaveLength(0);
        expect(wrapper.find('Grant')).toHaveLength(0);
        expect(wrapper.find('FAccountRequest')).toHaveLength(0);
        expect(wrapper.find('FEditGrant')).toHaveLength(0);
        expect(wrapper.find('FDashboard')).toHaveLength(0);
        expect(wrapper.find('FRequestSent')).toHaveLength(0);
        expect(wrapper.find('FSettings')).toHaveLength(0);
        expect(wrapper.find('SignIn')).toHaveLength(0);
        expect(wrapper.find('SignUp')).toHaveLength(0);
        expect(wrapper.find('ForgotPassword')).toHaveLength(0);
        expect(wrapper.find('MetaAdmin')).toHaveLength(0);
        expect(wrapper.find('FStripeSetup')).toHaveLength(0);
    });

    it('valid DDashboard path', () => {
        const wrapper = enzyme.mount(
            <MemoryRouter initialEntries={['/']}>
                {Routes}
            </MemoryRouter>
        );
        //expect(wrapper.containsMatchingElement(DDashboard)).toBeTruthy();

        expect(wrapper.find(DDashboard)).toHaveLength(0);
    });

})