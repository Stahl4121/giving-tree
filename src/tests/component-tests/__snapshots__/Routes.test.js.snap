// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes basics Renders correctly 1`] = `ShallowWrapper {}`;
